core/state/access_witness.go

```diff
@@ -159,7 +163,10 @@ func (aw *AccessWitness) TouchTxExistingAndComputeGas(targetAddr []byte, sendsVa
 	} else {
 		gas += aw.TouchAddressOnReadAndComputeGas(targetAddr, zeroTreeIndex, utils.BalanceLeafKey)
 	}
-	return gas
+	// Kaustinen note: we're currently experimenting with stop chargin gas for the origin address
+	// so simple transfer still take 21000 gas. This is to potentially avoid breaking existing tooling.
+	// This is the reason why we return 0 instead of `gas`.
+	return 0
```

Comment:
This method (and the one in my previous comment) are only used in the exact place we want this change: the top-level message wrapping the tx.

If that wasn't the case, clearly we couldn't make the change we wanted at the access witness level. But, since that's the case, I could avoid touching the main tx logic and take the decision of not charging the cost at this level which seams clearer to me.

Reply:
yeah that makes sense, I would just like to quell the linter warning